src/main/java/com/digitaldan/jomnilinkII/examples/Main.java

```diff
@@ -123,121 +121,259 @@ public void notConnectedEvent(Exception e) {
+					logger.info(thermos[i].toString());
+				}
+				status = c.reqObjectStatus(Message.OBJ_TYPE_THERMO, 1, thermoNum, true);
+				ExtendedThermostatStatus[] thermosExt = (ExtendedThermostatStatus[]) status.getStatuses();
+				for (int i = 0; i < thermosExt.length; i++) {
+					logger.info(thermosExt[i].toString());
+				}
 			}
-			status = c.reqObjectStatus(Message.OBJ_TYPE_AREA, 1, max_areas);
-			AreaStatus[] areas = (AreaStatus[]) status.getStatuses();
-			for (int i = 0; i < areas.length; i++) {
-				logger.info(areas[i].toString());
+			if (messageNum > 0) {
+				ObjectStatus status = c.reqObjectStatus(Message.OBJ_TYPE_MESG, 1, messageNum);
+				MessageStatus[] megs = (MessageStatus[]) status.getStatuses();
+				for (int i = 0; i < megs.length; i++) {
+					logger.info(megs[i].toString());
+				}
+				status = c.reqObjectStatus(Message.OBJ_TYPE_MESG, 1, messageNum, true);
+				ExtendedMessageStatus[] megsExt = (ExtendedMessageStatus[]) status.getStatuses();
+				for (int i = 0; i < megsExt.length; i++) {
+					logger.info(megsExt[i].toString());
+				}
 			}
-			status = c.reqObjectStatus(Message.OBJ_TYPE_THERMO, 1, max_thermos);
-			ThermostatStatus[] thermos = (ThermostatStatus[]) status.getStatuses();
-			for (int i = 0; i < thermos.length; i++) {
-				logger.info(thermos[i].toString());
+			if (auxNum > 0) {
+				ObjectStatus status = c.reqObjectStatus(Message.OBJ_TYPE_AUX_SENSOR, 1, auxNum);
+				AuxSensorStatus[] auxs = (AuxSensorStatus[]) status.getStatuses();
+				for (int i = 0; i < auxs.length; i++) {
+					logger.info(auxs[i].toString());
+				}
+				status = c.reqObjectStatus(Message.OBJ_TYPE_AUX_SENSOR, 1, auxNum, true);
+				ExtendedAuxSensorStatus[] auxsExt = (ExtendedAuxSensorStatus[]) status.getStatuses();
+				for (int i = 0; i < auxsExt.length; i++) {
+					logger.info(auxsExt[i].toString());
+				}
 			}
-			status = c.reqObjectStatus(Message.OBJ_TYPE_MESG, 1, max_mesgs);
-			MessageStatus[] megs = (MessageStatus[]) status.getStatuses();
-			for (int i = 0; i < megs.length; i++) {
-				logger.info(megs[i].toString());
+			if (audioZoneNum > 0) {
+				ObjectStatus status = c.reqObjectStatus(Message.OBJ_TYPE_AUDIO_ZONE, 1, audioZoneNum);
+				AudioZoneStatus[] audiozs = (AudioZoneStatus[]) status.getStatuses();
+				for (int i = 0; i < audiozs.length; i++) {
+					logger.info(audiozs[i].toString());
+				}
+				status = c.reqObjectStatus(Message.OBJ_TYPE_AUDIO_ZONE, 1, audioZoneNum, true);
+				ExtendedAudioZoneStatus[] audiozsExt = (ExtendedAudioZoneStatus[]) status.getStatuses();
+				for (int i = 0; i < audiozsExt.length; i++) {
+					logger.info(audiozsExt[i].toString());
+				}
 			}
-			status = c.reqObjectStatus(Message.OBJ_TYPE_AUX_SENSOR, 1, max_zones);
-			AuxSensorStatus[] auxs = (AuxSensorStatus[]) status.getStatuses();
-			for (int i = 0; i < auxs.length; i++) {
-				logger.info(auxs[i].toString());
+			if (expNum > 0) {
+				ObjectStatus status = c.reqObjectStatus(Message.OBJ_TYPE_EXP, 1, expNum);
```

Comment:
So this is throwing an exception since i don't have an expansion and `expNum` is hardcoded to "8" above in the file, `consoleNum` is also hardcoded

Reply:
That is because they both don't return valid responses for an object properties request. Any ideas?

Reply:
@digitaldan I commented out the code but left it there as an example, is that acceptable?